### ISSUE TITLE
Airplay volume changes not updated in MA frontend

### ISSUE THREAD

**Issue by Geweldich** [music-assistant/support#2823]
### What version of Music Assistant has the issue?

2.2.2

### What version of the Home Assistant Integration have you got installed?

none

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

When using Airplay (and perhaps other providers) the volume in MA/the MA frontend is not changed when the volume is changed externally (by a different application/controller/remote).

In my situation: when changing the volume in HA or the Devialet app, both immediately show the changed volume. In MA the volume doesn't change (even after waiting quite a long time).

### How to reproduce

1. Use HA or a different external method to change the volume of the device.

### Music Providers

Not relevant (happens with all)

### Player Providers

Airplay.

### What version of Home Assistant Core are your running

2024.8.3

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by marcelveldt** [comment]
Music Assistant is an AirPlay 1 (RAOP) streamer, not AirPlay 2. This means that only while streaming (and if the device implements it well), volume changes will be sent from the device to MA. When MA is not streaming to an Airplay device, nothing will happen.

**Comment by Geweldich** [comment]
I don’t completely understand. I am playing music from MA to a Devialet Phantom II. The Devialet app and Devialet HA integration both recognize each others volume changes and the volume changes done by MA. If I’m not mistaken volume changes can also be seen and done on Apple devices in the network.

**Comment by OzGav** [comment]
@Geweldich I believe what Marcel is saying is that those Devialet devices are probably using Airplay 2. MA only supports Airplay 1 which means that volume changes are only sent to MA when MA is streaming music to the device.

**Comment by OzGav** [comment]
2.2.3 has been released. Please confirm this has been fixed.

**Comment by Geweldich** [comment]
The situation has not changed for the Devialet Phantom II in my office. I noticed that when using uPNP volume changes do get detected and when using other AirPlay devices, volume changes do get reported back uniformly. So I guess Devialet cut some corners?

**Comment by marcelveldt** [comment]
I am afraid so but what you can do is the following;

In the settings for the airplay provider, set the log level to debug.
Now start playing something to the player.
Control the volume from the device itself a couple of times
Now download the full logfile from MA settings --> core modules and attach the file here

Maybe the device is sending some undocumented DACP command.

**Comment by Geweldich** [comment]
I will be back home tomorrow. I’ll get back to you.

**Comment by Geweldich** [comment]
Have not forgotten this. Have been having an outrageously long internet outage, which hopefully will get fixed this week. I’m limiting my phone/hotspot internet usage due to obvious related costs.

**Comment by Geweldich** [comment]
[music-assistant.log](https://github.com/user-attachments/files/17282982/music-assistant.log)
I enabled Airplay debugging and changed the device volume a couple of times (2 times up, 2 times down, 2 times up). Neither of these commands is visible in the attached log. So I guess we can close this issue without a fix?

**Comment by OzGav** [comment]
@marcelveldt thoughts?

**Comment by marcelveldt** [comment]
Yes, this is a player issue. It doesn't forward the volume adjustments as airplay DACP command.

There is however some hope for you:

1) We have plans to also add airplay 2 support at some point. Maybe the device's airplay 1 implementation is buggy but airplay 2 isn't.

2) We are going to add an option soon where you can specify a separate volume control for your MA players. So in that case you could redirect the volume to the HA control entity.

**Comment by charlesomer** [comment]
Hi, I think I am seeing the same or a similar issue with a few devices, Sonos, Edifier and a Yamaha amp specifically tested so far. When playing music via Music Assistant I can't change the volume from the Sonos app or Home Assistant for example. When I play music from my iPhone it works fine, changing the volume on the Sonos app is reflected on the phone. The Yamaha amp is AP1 only and works fine from the phone too so I'm not sure this is an AP1/AP2 difference specifically. Music Assistant doesn't seem to see any volume changes, I believe this used to work as I have some panels in Home Assistant which let me update the volume of the devices whilst Music Assistant is playing.

**Comment by charlesomer** [comment]
The volume gets reported to Music Assistant when Music Assistant is paused, it seems to only be when Music Assistant is actually playing something that the volume control doesn't work

**Comment by OzGav** [comment]
Reassigning the power and volume controls to a HA entity has been implemented in 2.4.0 https://beta.music-assistant.io/player-support/#player-controls @Geweldich please see if use of this will give you the functioanlity you need with this device.

**Comment by Geweldich** [comment]
Nice! I'm not running the beta version, so I'll check this change out once this hits stable.

**Comment by OzGav** [comment]
@Geweldich 2.4.0 has been released. Please confirm this is fixed

**Comment by OzGav** [comment]
We will close this soon assuming fixed
